tests/function/test_nontype.cpp

```diff
@@ -8,6 +8,9 @@ namespace
 
 struct A
 {
+    A() = default;
+    A(int a) : val(a) {}
```

Comment:
What errors did you see?

Reply:
This is the same clang problem as above, only this time is gives the following error:

~~~
error: no matching function for call to 'construct_at'

tests/function/test_nontype.cpp:109:41: note: in instantiation of function template specialization 'std::make_shared<(anonymous namespace)::A, int>' requested here
                        auto ptr = std::make_shared<A>(6);
~~~

Again, it fails to find the memberwise constructor of the struct. I also wrapped this in `#ifdef __clang__`

Reply:
Ah, I forgot this one. It's a different problem, though. I'll explain later.